scripts/autotracking.lua

```diff
@@ -0,0 +1,416 @@
+
+        local flagTest = value & flag
+
+        if flagTest ~= 0 and currentStage == 0 then
+            item.CurrentStage = 1
+            currentStage = 1
+        elseif currentStage == 1 then
+            item.CurrentStage = 0
+        end
+    end
+end 
+
+function updateSectionChestCountFromByteAndFlag(segment, locationRef, address, flag, callback)
+    local location = Tracker:FindObjectForCode(locationRef)
+    if location then
+        -- Do not auto-track this the user has manually modified it
+        if location.Owner.ModifiedByUser then
+            return
+        end
+
+        local value = ReadU8(segment, address)
+        
+        if AUTOTRACKER_ENABLE_DEBUG_LOGGING then
+            print(locationRef, value)
+        end
+  
+        if (value & flag) ~= 0 then
+            location.AvailableChestCount = 0
+            if callback then
+                callback(true)
+            end
+        else
+            location.AvailableChestCount = location.ChestCount
+            if callback then
+                callback(false)
+            end
+        end
+    elseif AUTOTRACKER_ENABLE_DEBUG_LOGGING then
+        print("Couldn't find location", locationRef)
+    end
+end
+
+function update2ItemLocation(segment, locationRef, item1Address, item1Flag, item2Address, item2Flag)
+  local location = Tracker:FindObjectForCode(locationRef)
+
+  if location then
+        -- Do not auto-track this the user has manually modified it
+        if location.Owner.ModifiedByUser then
+            return
+        end
+
+        local item1Value = ReadU8(segment, item1Address)
+        local item2Value = ReadU8(segment, item2Address)
+        local item1Retrieved = (item1Value & item1Flag) ~= 0
+        local item2Retrieved = (item2Value & item2Flag) ~= 0
+
+        if item1Retrieved and item2Retrieved then
+            location.AvailableChestCount = 0
+        elseif item1Retrieved or item2Retrieved then
+            location.AvailableChestCount = 1
+        else
+            location.AvailableChestCount = 2
+        end
+    elseif AUTOTRACKER_ENABLE_DEBUG_LOGGING then
+        print("Couldn't find locationRef ", locationRef)
+    end
+end
+
+function updateRushSoul(segment)
+  local item = Tracker:FindObjectForCode("rush")
+  if item then
+    local manitcoreValue = ReadU8(segment, 0x0201335D)
+    local manticoreTest = manitcoreValue & 0x0F
+    local curlyValue = ReadU8(segment, 0x0201335D)
+    local curlyTest =  curlyValue & 0xF0
+    local devilValue = ReadU8(segment, 0x0201335C)
+    local devilTest = devilValue & 0xF0
+    if manticoreTest ~= 0 or curlyTest ~= 0 or devilTest ~= 0 then
+      item.Active = true
+    else
+      item.Active = false
+    end
+  end
+end
+
+function updateItemsFromMemorySegment(segment)
+    if not isInGame() then
+        return false
+    end
+
+    InvalidateReadCaches()
+
+    if AUTOTRACKER_ENABLE_ITEM_TRACKING then
+        updateToggleItemFromByteAndFlag(segment, "djump", 0x02013393, 0x0F)
```

Comment:
Technically if they have exactly 16 of an item (or 32 or 64 etc) this will break but I feel like 16 skula's won't happen very often :P